cumulusci/tasks/salesforce/tests/test_install_package_version.py

```diff
@@ -78,33 +84,134 @@ def test_init_options():
+        project_config, get_resolver_stack(project_config, "production")
+    )
+
+
+@mock.patch(
+    "cumulusci.tasks.salesforce.install_package_version.GitHubDynamicDependency"
+)
+def test_init_options__dynamic_version_latest_beta(mock_GitHubDynamicDependency):
+    project_config = create_project_config()
+    project_config.config["project"]["package"]["namespace"] = "ns"
+
+    mock_GitHubDynamicDependency.return_value.managed_dependency = (
+        PackageNamespaceVersionDependency(namespace="ns", version="2.0 Beta 1")
+    )
+
     task = create_task(
         InstallPackageVersion,
         {"version": "latest_beta"},
         project_config=project_config,
     )
     assert task.options["version"] == "2.0 Beta 1"
 
+    mock_GitHubDynamicDependency.assert_called_once_with(github=project_config.repo_url)
+    mock_GitHubDynamicDependency.return_value.resolve.assert_called_once_with(
+        project_config, get_resolver_stack(project_config, "include_beta")
+    )
+
+
+@mock.patch(
+    "cumulusci.tasks.salesforce.install_package_version.GitHubDynamicDependency"
+)
+@mock.patch("cumulusci.tasks.salesforce.install_package_version.find_previous_release")
+def test_init_options__dynamic_version_previous(
+    mock_find_previous_release, mock_GitHubDynamicDependency
+):
+    project_config = create_project_config()
+    project_config.config["project"]["package"]["namespace"] = "ns"
+
+    mock_find_previous_release.return_value.tag_name = "release/1.0"
+    mock_GitHubDynamicDependency.return_value.managed_dependency = (
+        PackageNamespaceVersionDependency(namespace="ns", version="1.0")
+    )
+    project_config.get_repo = mock.Mock()
+
     task = create_task(
         InstallPackageVersion,
         {"version": "previous"},
         project_config=project_config,
     )
     assert task.options["version"] == "1.0"
 
+    mock_find_previous_release(
+        project_config.get_repo.return_value,
+        project_config.project__git__prefix_release,
+    )
+    mock_GitHubDynamicDependency.assert_called_once_with(
+        github=project_config.repo_url, tag="release/1.0"
+    )
+    mock_GitHubDynamicDependency.return_value.resolve.assert_called_once_with(
+        project_config, get_resolver_stack(project_config, "production")
+    )
+
+
+@mock.patch(
+    "cumulusci.tasks.salesforce.install_package_version.GitHubDynamicDependency"
+)
+def test_init_options__dynamic_version_no_managed_release(mock_GitHubDynamicDependency):
+    project_config = create_project_config()
+    project_config.config["project"]["package"]["namespace"] = "ns"
+
+    mock_GitHubDynamicDependency.return_value.managed_dependency = None
+
+    with pytest.raises(CumulusCIException) as e:
+        create_task(
+            InstallPackageVersion,
+            {"version": "latest_beta"},
+            project_config=project_config,
+        )
+
+        assert "does not identify a managed package" in str(e)
```

Comment:
This assertion is out of date, and not running because create_task raises the expected exception.

If you want to make an assertion about the exception message, do:

`with pytest.raises(CumulusCIException, match="some regexp"):`

Reply:
Whoops! Fixed.